src/game/rendering_graph_node.c

```diff
@@ -698,7 +706,11 @@ void geo_process_camera(struct GraphNodeCamera *node) {
     gCurLookAt->l[1].l.dir[0] = (s8)(127.0f * -(*cameraMatrix)[0][1]);
     gCurLookAt->l[1].l.dir[1] = (s8)(127.0f * -(*cameraMatrix)[1][1]);
     gCurLookAt->l[1].l.dir[2] = (s8)(127.0f * -(*cameraMatrix)[2][1]);
-#endif // F3DEX_GBI_2
+#endif
+
+#ifdef F3DEX_GBI_3
+        gSPCameraWorld(gDisplayListHead++, camWorld);
```

Comment:
Is this HLE safe?

Reply:
All new commands are noops for F3DEX2. Regardless, I don't think designing everything around faking the microcode in an emulator is a good idea.

Reply:
I would agree, but people **_WILL_** attempt to play hacks on incompatible plugins no matter what.

Reply:
If a hack creator is concerned about that, then F3DEX3 is not for them.

Reply:
> If a hack creator is concerned about that, then F3DEX3 is not for them.

To me at least, it's a moot point since you could say the same thing about libpl or libcart as well.

Reply:
libpl has checks in place to at least allow the user to design around both scenarios. Ideally, F3DEX3 should have some sort of runtime check that allows people to selectively do a thing depending on whether it's being used or not.

Reply:
There is only a single check I can imagine would work, which would be reading the results of the built in F3DEX3 profiler. I'd expect F3DEX2 HLE to leave the data as zeros.